### PR TITLE
Add reference to Sending Events to JavaScript

### PR DESCRIPTION
**Pull request by januswel**
Explain the **motivation** for making this change. What existing problem does the pull request solve?

I wondered to find ways to send data twice or more.

### CODE REVIEW

**Comment by rh389** [comment]
:+1: It's a good idea to make people aware of events in this section. My only reservation is that events aren't just about "twice or more" - often they're appropriate even if they never fire more than once (`onReady`, etc), due to the flexibility of allowing any number of listeners to opt-in.

How about "For an alternative mechanism for sending data to JavaScript, see"...?

**Comment by januswel** [comment]
@rh389 Thank you for your suggestions!
I think below description makes developers "How I send data twice or more to show progresses?"

> A native module is supposed to invoke its callback only once

So I added descriptions about capabilities and exceptions.

**Comment by januswel** [comment]
How about this?

File: docs/NativeModulesIOS.md [hunk 1]
```diff
@@ -180,6 +180,8 @@ CalendarManager.findEvents((error, events) => {
 
 A native module is supposed to invoke its callback only once. It can, however, store the callback and invoke it later. This pattern is often used to wrap iOS APIs that require delegates. See [`RCTAlertManager`](https://github.com/facebook/react-native/blob/master/React/Modules/RCTAlertManager.m) for an example.
 
+For an alternative mechanism for sending data to JavaScript, see [Sending Events to JavaScript](#sending-events-to-javascript). You can use the mechanism to send data twice or more, except for some events should be fired once, for example, `onReady`.
```
Comment (mkonicek) [inline docs/NativeModulesIOS.md]: `#sending-events-to-javascript` seems to be invalid? I can't find it in the project.

**Comment by mkonicek** [comment]
Thanks for the pull request. However, IMHO it's clear that an event can be used to send the same data twice. Second, it's not clear what `onReady` in this example means - does it have to do with networking? I don't seen 'onReady' mentioned elsewhere in the document so not sure what it is.

**Comment by rh389** [comment]
Apologies @januswel - I added to the confusion by inventing onReady as a hypothetical example of an exactly-once event, I didn't mean to suggest incorporating it into the PR. Ah well.

**Comment by januswel** [comment]
`#sending-events-to-javascript` points to https://facebook.github.io/react-native/docs/native-modules-ios.html#sending-events-to-javascript

> IMHO it's clear that an event can be used to send the same data twice

I agree with this, but in a section "Callbacks", below line makes developers confused.  Because callbacks are called more than once, generally. 

> A native module is supposed to invoke its callback only once. 

And, the section "Sending Events to Callbacks" is separated from "Callbacks" and "Promises". This also makes difficult to find a way to send data twice or more.

**Pull request closed without merging**